### PR TITLE
Move DependencyResolver to a separate library

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Move the logic from `InstanceCreator` and `DependencyResolver` to a new and decoupled library: 
`gacela-project/dependency-resolver` ([packagist](https://packagist.org/packages/gacela-project/dependency-resolver))

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
     "prefer-stable": true,
     "minimum-stability": "dev",
     "require": {
-        "php": "^8.0, <8.3"
+        "php": "^8.0, <8.3",
+        "gacela-project/dependency-resolver": "^0.1"
```
Comment (JesusValera) [inline composer.json:31]: Thinking about the package name... 🤔 

Some proposals:
- `Dependency Resolver`
- `Dependency Container`
- `Dependency Injector`
- `Service Container` 
- `Container` -> [Laravel](https://laravel.com/docs/10.x/container)
- `Dependency Injection` -> [Symfony](https://symfony.com/doc/current/components/dependency_injection.html)
- `Object Resolver` (was the name proposed by @Chemaclass?)
- you say ...!

I think I like `Dependency Resolver`, but open to discuss.
But I think we should add these proposals to the `keywords` key on L13 to have more visibility, what do you think?

**Review by JesusValera — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (Tito-Kati) [reply composer.json:31]: Resolver ☝🏻


Another option would be Binder.

Comment (Chemaclass) [reply composer.json]: I like that. I renamed it to just "resolver" -> `gacela-project/resolver`

**Review by Tito-Kati — APPROVED** [review]
👌️

**Pull request merged**